### ISSUE TITLE
ClassCastException when using JTable in sub-window since 3.1

### ISSUE THREAD

**Issue by lilium724** [JFormDesigner/FlatLaf#722]
**Platform:** Windows 11

I got this error while using [Ghidra](https://github.com/NationalSecurityAgency/ghidra) with FlatLaf for the dark mode, it crashes with an error message whenever I open a sub-window with a JTable in it. It works fine with the 3.0 version, and since the stacktrace doesn't mention any ghidra method, i figured it might have been caused by 1435469ee5ce1c2b584d5e0a04e5cf5ebb7016ff

I'm trying to reproduce the issue in a smaller project, but so far I haven't found anything.

```
2023-08-27 13:44:37 ERROR (SwingExceptionHandler) Error: Uncaught Exception! 
ClassCastException - class com.formdev.flatlaf.ui.FlatTableHeaderUI$FlatTableCellHeaderRenderer cannot be cast to class
 java.awt.Component (com.formdev.flatlaf.ui.FlatTableHeaderUI$FlatTableCellHeaderRenderer is in unnamed module of loader
 'app'; java.awt.Component is in module java.desktop of loader 'bootstrap') java.lang.ClassCastException: class com.formdev.flatlaf.ui.FlatTableHeaderUI$FlatTableCellHeaderRenderer cannot be cast to class java.awt.Component (com.formdev.flatlaf.ui.FlatTableHeaderUI$FlatTableCellHeaderRenderer is in unnamed module of loader 'app'; java.awt.Component is in module java.desktop of loader 'bootstrap')
	at docking.widgets.table.GTableHeaderRenderer.setBounds(GTableHeaderRenderer.java:87)
	at java.desktop/javax.swing.CellRendererPane.paintComponent(CellRendererPane.java:156)
	at java.desktop/javax.swing.plaf.basic.BasicTableHeaderUI.paintCell(BasicTableHeaderUI.java:741)
	at java.desktop/javax.swing.plaf.basic.BasicTableHeaderUI.paint(BasicTableHeaderUI.java:683)
	at com.formdev.flatlaf.ui.FlatTableHeaderUI.paint(FlatTableHeaderUI.java:273)
	at java.desktop/javax.swing.plaf.ComponentUI.update(ComponentUI.java:161)
	at java.desktop/javax.swing.JComponent.paintComponent(JComponent.java:852)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1128)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:961)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1137)
	at java.desktop/javax.swing.JViewport.paint(JViewport.java:736)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:961)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1137)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:961)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1137)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:961)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1137)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:961)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1137)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:961)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1137)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:961)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1137)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:961)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1137)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:961)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1137)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:961)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1137)
	at java.desktop/javax.swing.JLayeredPane.paint(JLayeredPane.java:586)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:961)
	at java.desktop/javax.swing.JComponent.paintToOffscreen(JComponent.java:5325)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paintDoubleBufferedFPScales(RepaintManager.java:1720)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paintDoubleBuffered(RepaintManager.java:1629)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paint(RepaintManager.java:1569)
	at java.desktop/javax.swing.RepaintManager.paint(RepaintManager.java:1336)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1114)
	at java.desktop/java.awt.GraphicsCallback$PaintCallback.run(GraphicsCallback.java:39)
	at java.desktop/sun.awt.SunGraphicsCallback.runOneComponent(SunGraphicsCallback.java:75)
	at java.desktop/sun.awt.SunGraphicsCallback.runComponents(SunGraphicsCallback.java:112)
	at java.desktop/java.awt.Container.paint(Container.java:2005)
	at java.desktop/java.awt.Window.paint(Window.java:3959)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:889)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:861)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:861)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:834)
	at java.desktop/javax.swing.RepaintManager.prePaintDirtyRegions(RepaintManager.java:784)
	at java.desktop/javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1897)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

**Comment by lilium724** [comment]
Never mind, I didn't realize ghidra now uses it's own custom version of FlatLaf, and the TableHeaderRenderer classes aren't compatible.

**Comment by DevCharly** [comment]
You're right that the issue is caused by commit 1435469ee5ce1c2b584d5e0a04e5cf5ebb7016ff, which now uses a delegating header renderer that is not an instance of `java.awt.Component`.

The problem is that Ghidra class [GTableHeaderRenderer](https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Framework/Docking/src/main/java/docking/widgets/table/GTableHeaderRenderer.java) assumes that the renderer is an instance of `java.awt.Component`. There are several places where the field `TableCellRenderer delegate` is cast to `Component`. E.g.

https://github.com/NationalSecurityAgency/ghidra/blob/b261137b4093bc2cd9bf1f809a6c002e1b6d8d75/Ghidra/Framework/Docking/src/main/java/docking/widgets/table/GTableHeaderRenderer.java#L87

This needs to be fixed in Ghidra before upgrading to FlatLaf 3.1 or newer.
